src/NuGet.Clients/PackageManagement.VisualStudio/ProjectSystems/LegacyCSProjPackageReferenceProjectProvider.cs

```diff
@@ -52,6 +65,9 @@ public bool TryCreateNuGetProject(EnvDTE.Project dteProject, ProjectSystemProvid
                 return false;
             }
 
+            // Lazy load the CPS enabled JoinableTaskFactory for the UI.
+            NuGetUIThreadHelper.SetJoinableTaskFactoryFromService(_componentModel.Value);
```

Comment:
Why does the legacy csproj need the CPS factory? `GetUnconfiguredProject` in this file is never even called from what I can tell.